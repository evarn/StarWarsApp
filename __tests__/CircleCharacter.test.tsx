import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';
import Strings from '../src/constants/strings';
import CircleCharacter from '../src/screens/card/components/Circle';
import Colors from './../src/constants/colors';

describe('CircleCharacter', () => {
  it('CircleCharacter snapshoot', () => {
    const {toJSON} = render(
      <CircleCharacter
        title={'mass'}
        param={'50'}
        style={{backgroundColor: Colors.BLACK_1}}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('CircleCharacter text', () => {
    const {getByText} = render(
      <CircleCharacter
        title={'mass'}
        param={'50'}
        style={{backgroundColor: Colors.BLACK_1}}
      />,
    );
    const title = 'mass';
    const foundTextTitle = getByText(title);
    expect(foundTextTitle.props.children).toEqual(title);
  });
});
