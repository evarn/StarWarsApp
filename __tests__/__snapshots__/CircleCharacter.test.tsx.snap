// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleCharacter CircleCharacter snapshoot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "margin": 12,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderColor": "#000000",
          "borderRadius": 27,
          "borderWidth": 3,
          "height": 54,
          "justifyContent": "center",
          "width": 54,
        },
        {
          "backgroundColor": "#000000",
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "#212121",
          "fontFamily": "Karla",
          "fontSize": 15,
          "fontStyle": "normal",
          "fontWeight": "400",
        }
      }
    >
      50
    </Text>
  </View>
  <Text
    style={
      {
        "color": "#4B4B4B",
        "fontFamily": "Karla",
        "fontSize": 12,
        "fontStyle": "normal",
        "fontWeight": "400",
        "marginTop": 4,
        "textAlign": "center",
      }
    }
  >
    mass
  </Text>
</View>
`;
